docs/design/018-new-http-api.md

```diff
@@ -186,22 +188,44 @@ export default async function () {
 }
 ```
 
-- A tentative HTTP/3 example:
+- Configuring TLS with a custom CA certificate and forcing HTTP/2:
 ```javascript
-import { UDP } from 'k6/x/net';
+import { TCP } from 'k6/x/net';
 import { Client } from 'k6/x/net/http';
+import { open } from 'k6/x/file';
 
-export default async function () {
-  const socket = new UDP.open('192.168.1.1:9090');
+const caCert = await open('./custom_cacert.pem');
 
+export default async function () {
   const client = new Client({
-    socket: socket,
-    version: 3,      // A UDP socket would imply HTTP/3, but this makes it explicit.
+    dial: async address => {
+      return await TCP.open(address, {
+        tls: {
+          alpn: ['h2'],
+          caCerts: [caCert],
+        }
+      });
+    },
   });
-  await client.get('https://httpbin.test.k6.io/get');
+  await client.get('https://10.0.0.10/');
 }
 ```
 
+- Forcing unencrypted HTTP/2 (h2c):
+```javascript
+import { TCP } from 'k6/x/net';
+import { Client } from 'k6/x/net/http';
+
+export default async function () {
+  const client = new Client({
+    dial: async address => {
+      return await TCP.open(address, { tls: false });
+    },
+    version: [2],
```

Comment:
Should we have different implementations?

```js
import { Client } from 'k6/x/net/http';  // this is v1
import { Clientv1, Clientv2, Clientv3 } from 'k6/x/net/http';
```
Ofc, they can share the common parts under the hood but still give us flexibility in handling different logic.